spec/buffered_line_reader_spec.rb

```diff
@@ -0,0 +1,62 @@
+require "ld-eventsource/impl/buffered_line_reader"
+
+def tests_for_terminator(term, desc)
+  def make_tests(name, input_line_chunks, expected_lines)
+    [{
+      name: "#{name}: one chunk per line",
+      chunks: input_line_chunks,
+      expected: expected_lines
+    }].concat(
+      (1..4).map do |size|
+        ({
+          name: "#{name}: #{size}-character chunks",
+          chunks: input_line_chunks.join().chars.to_a.each_slice(size).map { |a| a.join },
```

Comment:
Fun thing with Ruby, `join` will go down collections:

```ruby
['a', 'b', ['c', ['d']]].join
# => "abcd"
```

So you could potentially do something like this:

```ruby
input_line_chunks.flat_map(&:chars).each_slice(size).join
```

Reply:
But that's not what I want. The result of this expression is not supposed to be a single string, but an array of strings.

The intention here is to test for edge cases that can result from lines or line terminators being split differently across chunks as the chunks are read into the buffer. If the original tests specifies `input_line_chunks` as `-`["line1\r\n", "line2\r\n"]`, we will test with exactly those chunks, but we'll also test with them split up differently: for instance when `size` is 2, we want the chunks to be delivered as `["li", "ne", "1\r", "\nl", "in", "e2", "\r\"]`.

Reply:
@louis-launchdarkly OK - the reason I didn't leave a comment is that the name of the test is already "2-character chunks", etc. which I thought described the intention of making 2-character chunks, but I can add one.